uiowa/uiowa
# add media_entity_file_replace, delete old orphaned files

Resolves #883 
Resolves #1867 

- [x] Create follow-up documentation regarding replace functionality, known issues - https://github.com/uiowa/uiowa/issues/2180
- [x] Get stats on how much disk space is cleaned up after the update hook runs (BEFORE: 69% PROD, 75% DEV/STAGE)

## To Test Media Entity File Replace

Try using the replace field on existing media entities (audio, file, image). If the media was referenced on content, that content should get the updated file/version.

File metadata (dimensions, size, etc.) might not be updated, because Media doesn't really know anything has changed. https://www.drupal.org/project/media_entity_file_replace/issues/3109553 which leads to core issues

## To Test Orphan File Deletion Update Hook

Pull down an old existing site with a lot of files. The update hook should mark status = 0 for at least a couple files in file_managed table. Run cron. Those entries should go away and your files directory should not have those files.

On a v2 site since the entity embed icon was probably deleted from file_managed, open up a WYSIWYG and make sure an icon loads. It should be recreated in your local directory but not in the files_managed table.

If you have a site that relied on non-Media reference fields, make sure you aren't getting unintended consequences with those fields (e.g. uiowa_footer lockups, brand site lockups, dcv files)
